core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -557,26 +593,36 @@ public void clearDataMaps(String tableUniqName) {
   public void clearDataMap(AbsoluteTableIdentifier identifier, String dataMapName) {
     CarbonTable carbonTable = getCarbonTable(identifier);
     String tableUniqueName = identifier.getCarbonTableIdentifier().getTableUniqueName();
-    List<TableDataMap> tableIndices = allDataMaps.get(tableUniqueName);
-    if (tableIndices != null) {
-      int i = 0;
-      for (TableDataMap tableDataMap : tableIndices) {
-        if (carbonTable != null && tableDataMap != null && dataMapName
-            .equalsIgnoreCase(tableDataMap.getDataMapSchema().getDataMapName())) {
-          try {
-            DataMapUtil.executeDataMapJobForClearingDataMaps(carbonTable);
-            tableDataMap.clear();
-          } catch (IOException e) {
-            LOGGER.error("clear dataMap job failed", e);
-            // ignoring the exception
+    if (CarbonProperties.getInstance()
```

Comment:
Can we disable index cache server property after enabling it and executing few queries?..If it is a possible scenario then how cache clean up is taken care from the distributed cache server for the above scenario?..Is the cache cleaned from index cache server while it being disabled?

Reply:
No, the cache would not be cleared because it can happen that in a second session some other user is accessing the querying the same  table using index server.

Reply:
So the cache will be cleaned up only during table/segment deletion...ryt?

Reply:
yes